### ISSUE TITLE
Add configuration support

### ISSUE THREAD

**Issue by CMDJojo** [modmark-org/modmark#149]
**Is your feature request related to a problem? Please describe.**
We need a way to configure the ModMark compiler, like adding new packages, hiding built-in packages, possibly adjusting parser settings etc etc. This needs to have support, and there is a couple of different ways to do it:

**Option 1: Add a new `[use]/[import]` native module**
We can add a new module called `[use]`, `[import]` or similar that takes an URL/path and loads that as a package. This approach is indeed an option, but there is a couple of problems with it:
 - We cannot serialise modules that aren't loaded (see #125, #145, #148).
     - This means that if we use a `[table]` module (assuming it isn't loaded), `core` must first evaluate the `[import] table.wasm` module.
     -  Not only does this require scheduling, at least to some primitive extent, but our current solution doesn't work since it evaluates outside-in, meaning that `__document` is evaluated before any `[import]` module is, meaning that `[table]` must be serialisable before `[import] table` is executed, OR an immediate rewrite of the scheduling is necessary.
     - Changing the serialisation algorithm is possible, albeit convoluted, since it currently maps all arguments to KV-pairs and we would need to have a workaround for non-loaded modules. This will lead to inconsistencies in serialisation depending on if a module is loaded or not, which I think is really bad.
     - We could change the evaluation order to be inside-out, for the time being, and change it even more later when we get a more advanced scheduling system in place.
 - Other packages may spit out `[use]/[import]` modules
     - While it might seem reasonable to allow other packages to import modules on a need-to-use basis, I think this may just fill out the outer scope with lots of different modules
     - This would of course require scoping, to be done in any decently controlled manner, but we have no scoping system in place at the moment, leading to this issue being blocked by the progress of variables/a scoping system
     - Additionally, even if scoped variables are added, it isn't clear that Context is necessarily scoped
     - One solution may be to, for the time being, only allow top-level `[use]/[import]`, but we have no way of knowing if a module is top-level or not. We would need to add an exception for this special use case
 - Having multiple `[use]` statements would make it harder to send a batch job (i.e *import module1,module2,module3*) to whatever system is resolving the packages. This might make it harder to make it run in parallell.

**Option 2: Add a fake `[config]` "parser" module**
We can add a new "fake" `[config]` module, which essentially is a special-case where the parser optionally searches for a `[config]` module right at the start of the document. If it finds one, it will parse it itself, yielding some kind of configuration object saying what imports should be used etc.
 - This would be simple to implement, we could possibly just use the same parsers as we are using now plus nom:s `verify` to make sure it is a `[config]` module
     - The syntax rules of the `[config]` module would have to be housed in the parser itself which is both good and bad.
         - On the good side, it would give faster error reporting, and `nom` is already a good parser library which would allow for good, structured syntax.
         - On the bad side, it means that other packages may not spit out such `[config]` modules. This is both good and bad in some sense, it would be possible to know all packages to load at one time instead of one-by-one, and if a package depends on other packages, it might be able to say so in the manifest
     - It would be able to configure the parser itself, for example turning off smart punctuation, possibly adding/removing tags etc. It isn't completely certain if we would want to allow for this anyways, but using this option, it is still possible while with option 1, it wouldn't be that easily.
 - It would be a kind of work-around just for one special module
 - It would be able to, together with the AST of the document, give the configuration as a struct directly to `core` for it to use, without `core` having to worry about syntax and such.
 - We could, by reading one `[config]` module, know every declared import which makes it easy to do as a batch job.

**Describe the solution you'd like**
I would like to implement **Option 2** since I think it is more easily done in the current state of the project. If **Option 1** is to be used instead, this might be changed to later on, but I think there is too much in the way to currently implement **Option 1** without too many workarounds. I think **Option 2** would be pretty easy to implement. Additionally, I think it might be fun to at least leave customisable tags open.

**Describe alternatives you've considered**
Me and @adelhult have discussed a lot, and while I am leaning towards **Option 2**, he is leaning towards **Option 1**. I would love any input and feedback that may deepen the discussion

**Additional context**
We would of course try to make the syntax of the module somewhat intuitive. One example syntax would be:
```
[config]
[use] table
[use] html hiding __bold
[use] latex using __italics
[hide] code
```
If you have any syntax ideas, please send them!

**Comment by adelhult** [comment]
I think that **option 1** has the potential of being a much more elegant solution _in theory_. But, as mentioned it would require a lot of other systems before even being remotely functional. So I agree that we should opt for **option 2** which is a bit more pragmatic and likely would work without that much trouble. It would also allow us to be flexible to expand or change  the system later if need be.


# Naming
We may want to consider calling it `[config]`, `[configuration]`, `[settings]`, `[options]` or `[imports]` something else? I guess that we should go for a name that non-technical people are familiar with (imports may be a bad choice if we want to include other parsing settings too).

I think that `[config]` should work just fine, but may be worth thinking a bit about. 

# Import syntax
I think that we don't need to include brackets around the verbs inside of the modules, Since they are not actually modules I think that we might as well skip making them look like it and instead go for a more readable and expressive syntax. We may want to take inspiration from the import syntax found in ML-style languages like Haskell and Elm. And perhaps Python and Julia too.

Maybe something like this:
```haskell
[config]
-- brings all the modules into scope
import local_pkg  
import https://eli.nu/online_package.wasm

 -- brings only the specified modules into scope
import local_pkg2 (foo, bar, baz)  

-- brings everything into scope except the mentioned modules
import pkgs:adelhult/coolmath hiding (math, bettermath)  
```
(Note that we would need somewhat more complicated syntax, we may for instance want to have `as` to rename modules and we may also want to expose `[math]->latex` but not `[math]->html`. )

# Config files
Another approach that @Forsinge mentioned is that we may want to support configurations specified in a external file too. In that case we may instead want the syntax to be valid [TOML](https://toml.io/en/)  (or YAML) instead.

**Comment by CMDJojo** [comment]
Closed by #169
